### ISSUE TITLE
Admin images and folder empty, data still present

### ISSUE THREAD

**Issue by RRMoelker** [divio/django-filer#438]
I've run into a problem where the filer admin files and folders are no longer visible. I traced it back to a change from version 0.9.5 to 0.9.6.

This is how the admin looks in 0.9.5:
![filer_0 9 5](https://cloud.githubusercontent.com/assets/205326/4029307/d03050f4-2c4d-11e4-9aca-34386fdb2d2d.png)
(edited image to remove user data)

This is how the admin looks in 0.9.6:
![filer_0 9 6](https://cloud.githubusercontent.com/assets/205326/4029314/d3a90e06-2c4d-11e4-8a61-dfd3829e1eed.png)
Note that in the second image the folders are no longer showing visible, but it does however state that there are 4 folders in the view.

I've encountered the issue on two different machines and disabling django-suit didn't seem to solve te problem. A set of the installed packages for the *none* working setup:

Django==1.5.9
*django-filer==0.9.6*
django-suit==0.2.9
django-autocomplete-light==1.4.14
South
django-compressor (only used for css in admin)

The autocomplete light javascript throws an error "Uncaught TypeError: undefined is not a function widget.js:343". But it does so in both the working and broken version. I think this is unrelated and is caused by loading jquery in the wrong order (jquery, autocomplete.js, jquery, filer.js).

Using an older version is a good enough fix at the moment for us, but I hope this helps somebody else.

**Comment by yunmanger1** [comment]
![screenshot from 2016-03-14 16 50 12](https://cloud.githubusercontent.com/assets/388285/13742254/128d500c-ea05-11e5-90da-fe1917ead80f.png)

Having this issue with `django-cms==3.2.1` and `django==1.8.7`

**Comment by yunmanger1** [comment]
@stefanfoulis 

I had hours of fun debugging. I almost got defeated by this bug.

These issues are not solved completely  #753 #630

This line below ruins everything in production when `ManifestStaticFilesStorage` is used and especially when template debugging is off.

https://github.com/divio/django-filer/blob/develop/filer/models/mixins.py#L20

The `admin/filer/folder/directory_listing.html` includes `admin/filer/folder/directory_table.html` which uses icons on the model. The most interesting part is that when template debugging is disabled, no errors happen. And on local machine I have different static storage. Once I enabled template debugging on production I could see an error:

```
ValueError: The file 'filer/icons/unfiled_folder_16x16.png' could not be found with <sdu_site.contrib.static.PipelineGzipManifestWhitenoiseStorage object at 0x7fc016485190>.
```

**Comment by stefanfoulis** [comment]
@yunmanger1 thanks for the analysis!

**Comment by Chronial** [comment]
@yunmanger1 I see you are using a custom static storage. It seems likely that this bug is not reproducible without it.
Could you reduce it to the minimum that triggers this bug?

This might also just be a bug in your custom storage class. The file it can't find is definitely there, so it should find it. I think the django-filer code you are referring to has the property of calling `static()` unusually early in the request, before template rending started. Maybe your storage class isn't set up properly at this point?

If this is the case it might still be worth changing django-filer, because this unusual behaviour of django-filer is surely not helping :).

**Comment by yunmanger1** [comment]
I know exactly where it happens. 

https://github.com/django/django/blob/master/django/contrib/staticfiles/storage.py#L93

If you use `django.contrib.staticfiles.storage.ManifestStaticFilesStorage` as a storage.

Particularly in my case I am using this:

```
from whitenoise.django import GzipManifestStaticFilesStorage
from pipeline.storage import PipelineMixin


class PipelineGzipManifestWhitenoiseStorage(PipelineMixin, GzipManifestStaticFilesStorage):
    pass
```

`FILER_ADMIN_ICON_SIZES` by default has `('16', '32', '48', '64')` values, `unfiled_folder_16x16.png` is not in icons folder.

Actually @stefanfoulis removed the '16' from default icon sizes nine days ago in 1089ca1, 
but this is not solution because there are other icons that are not present, for example `missingfile_64x64.png` or no file `nofile_64x64.png` which might be not used in template, but the IconsMixin will accessed via `static` because '64' is present in `FILER_ADMIN_ICON_SIZES`

**Comment by stefanfoulis** [comment]
ok, lets add those missing icons then. @mikek could you have a look at this please?

**Comment by mikek** [comment]
@stefanfoulis, after some debugging It seems that to fix this adding missing icons (sizes) should be enough. I suggest we even catch/ignore `ValueError` around the [static()] (https://github.com/divio/django-filer/blob/f534131ba48c52405f00e8779d3bfefcf2cebda0/filer/models/mixins.py#L20) call in `IconsMixin` for the case when there are still some missing icons. 

It does not affect `collectstatic` command and is not strictly related to #630.

**Comment by stefanfoulis** [comment]
ok :+1:

**Comment by mikek** [comment]
Fixed in #785
